--- origsrc/qrencode-3.0.3/Makefile.am	2008-05-26 12:57:42.000000000 -0500
+++ src/qrencode-3.0.3/Makefile.am	2008-06-17 21:45:41.718750000 -0500
@@ -15,7 +15,7 @@
 						 rscode.c rscode.h \
 						 split.c split.h \
 						 mask.c mask.h
-libqrencode_la_LDFLAGS = -version-number $(MAJOR_VERSION):$(MINOR_VERSION):$(MICRO_VERSION)
+libqrencode_la_LDFLAGS = -version-number $(MAJOR_VERSION):$(MINOR_VERSION):$(MICRO_VERSION) -no-undefined
 
 include_HEADERS = qrencode.h
 
@@ -30,6 +30,6 @@
 bin_PROGRAMS = qrencode
 qrencode_SOURCES = qrenc.c
 qrencode_CFLAGS = $(png_CFLAGS)
-qrencode_LDADD = -lqrencode $(png_LIBS)
+qrencode_LDADD = libqrencode.la $(png_LIBS)
 man1_MANS = qrencode.1
 endif
